app/models/user.rb

```diff
@@ -716,6 +716,14 @@ def self.current
     @current_user ||= User.anonymous
   end
 
+  def self.execute_as(user)
```

Comment:
While I see that this existed before this PR and thus may be out of scope, I really dislike this approach. Why exactly is this override necessary? I couldn't detect anything in the used UserMailer methods. Can you provide me with some insights as to why this remains necessary?